src/components/modes/teacher/Settings.js

```diff
@@ -142,9 +159,12 @@ class Settings extends Component {
 
   handleChangeProgrammingLanguage = ({ target }) => {
     const { value } = target;
-    this.setState({
-      selectedProgrammingLanguage: value,
-    });
+    this.setState({ ...this.state, selectedProgrammingLanguage: value });
+  };
+
+  handleChangeVisibility = ({ target }) => {
+    const { value } = target;
+    this.setState({ ...this.state, selectedVisibility: value });
```

Comment:
There's a better way of doing this, so that you don't have to remove the ESLint rule.

https://www.freecodecamp.org/news/understanding-setstate-in-react-ea8982168b49/

Read the article above. The solution is under the phrase: *"And add a new click handler onClickfn() as shown below"*